consensus/mock/delayedBroadcasterMock.go

```diff
@@ -0,0 +1,55 @@
+package mock
+
+import (
+	"github.com/multiversx/mx-chain-core-go/data"
+
+	"github.com/multiversx/mx-chain-go/consensus/broadcast"
+)
+
+type DelayedBroadcasterMock struct {
+	SetLeaderDataCalled         func(data *broadcast.DelayedBroadcastData) error
+	SetValidatorDataCalled      func(data *broadcast.DelayedBroadcastData) error
+	SetHeaderForValidatorCalled func(vData *broadcast.ValidatorHeaderBroadcastData) error
+	SetBroadcastHandlersCalled  func(mbBroadcast func(mbData map[uint32][]byte, pkBytes []byte) error, txBroadcast func(txData map[string][][]byte, pkBytes []byte) error, headerBroadcast func(header data.HeaderHandler, pkBytes []byte) error) error
+	CloseCalled                 func()
+}
+
+func (d DelayedBroadcasterMock) SetLeaderData(data *broadcast.DelayedBroadcastData) error {
```

Comment:
we should use pointers here, thus `d DelayedBroadcasterMock` should become `mock *DelayedBroadcasterMock`. Valid for all methods

Reply:
updated